exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ScanBatch.java

```diff
@@ -173,9 +174,8 @@ public IterOutcome next() {
 
         currentReader.allocate(mutator.fieldVectorMap());
       } catch (OutOfMemoryException e) {
-        logger.debug("Caught Out of Memory Exception", e);
         clearFieldVectorMap();
-        return IterOutcome.OUT_OF_MEMORY;
+        throw UserException.memoryError(e).build(logger);
```

Comment:
Makes sense.

Asking the question a different way.. To avoid regressions, should the pre-requisite changes ([DRILL-5211](https://issues.apache.org/jira/browse/DRILL-5211) and pertinent tasks) be committed before this patch? Or since _the readers do not correctly handle the case_ anyway, there will be no difference?